tidy3d/plugins/smatrix/ports/base_lumped.py

```diff
@@ -0,0 +1,101 @@
+"""Class and custom data array for representing a scattering matrix port based on lumped circuit elements."""
+
+import pydantic.v1 as pd
+from typing import Optional
+from abc import ABC, abstractmethod
+
+from ....constants import OHM
+from ....components.types import Complex, FreqArray
+from ....components.base import Tidy3dBaseModel, cached_property
+from ....components.grid.grid import Grid, YeeGrid
+from ....components.lumped_element import AbstractLumpedResistor
+from ....components.monitor import FieldMonitor
+
+from ....components.data.sim_data import SimulationData
+from ....components.source import UniformCurrentSource, GaussianPulse
+
+from ....components.data.data_array import DataArray, FreqDataArray
+
+DEFAULT_PORT_NUM_CELLS = 3
+
+
+class LumpedPortDataArray(DataArray):
+    """Port parameter matrix elements for lumped ports.
+
+    Example
+    -------
+    >>> port_in = ['port1', 'port2']
+    >>> port_out = ['port1', 'port2']
+    >>> f = [2e14]
+    >>> coords = dict(
+    ...     port_in=ports_in,
+    ...     port_out=ports_out,
+    ...     f=f
+    ... )
+    >>> fd = LumpedPortDataArray((1 + 1j) * np.random.random((2, 2, 1)), coords=coords)
+    """
+
+    __slots__ = ()
+    _dims = ("port_out", "port_in", "f")
+    _data_attrs = {"long_name": "lumped port matrix element"}
+
+
+class AbstractLumpedPort(Tidy3dBaseModel, ABC):
+    """Class representing a single lumped port"""
+
+    name: str = pd.Field(
+        ...,
+        title="Name",
+        description="Unique name for the port.",
+        min_length=1,
+    )
+
+    impedance: Complex = pd.Field(
+        50,
```

Comment:
might be good to make this a module constant, eg `DEFAULT_REFERENCE_IMPEDANCE = 50` after the imports.

This helps a bit with testing and documentation.